app/main.py

```diff
@@ -2,5 +2,17 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_history = {}
+
+    def wrapper(*args):
+        nonlocal cache_history
+
+        if args in cache_history.keys():
+            print("Getting from cache")
+            return cache_history[args]
+        else:
+            print("Calculating new result")
+            result = func(args)
+            cache_history["args"] = result
```

Comment:
The key used in `cache_history` should be `args` instead of the string "args". This will ensure that the actual arguments are used as the key for caching results.